### PR TITLE
125-show-alert-for-non-handled-files

### PR DESCRIPTION
**Pull request by alvrba**

### CODE REVIEW

File: src/components/ItemUnsupported.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { Text } from 'react-native-elements';
+
+const ItemUnsupported = () => {
+  const { t } = useTranslation();
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {t('This file cannot be opened in the mobile app')}
+      </Text>
```
Comment (spaenleh) [inline src/components/ItemUnsupported.tsx:12]: Maybe we can add a suggestion to contact us in order to request support for some file types ? This way we can reduce frustration and learn what we need to implement in priority.

File: src/components/ItemUnsupported.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { StyleSheet, View } from 'react-native';
+import { Text } from 'react-native-elements';
+
+const ItemUnsupported = () => {
+  const { t } = useTranslation();
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {t('This file cannot be opened in the mobile app')}
```
Comment (spaenleh) [inline src/components/ItemUnsupported.tsx:11]: Maybe display the item type here so we know what we need to support. 
I think `file` may not be the right wording. I would suggest using `element` instead, as they can be interactive elements (h5p, etherpads etc) and not strictly files.

**Review by spaenleh — CHANGES_REQUESTED** [review]
I think we can include contact information to request support of a type.

Also tests ? 👀

**Comment by alvrba** [comment]
Pending to first update to Expo SDK 50 and then add Sentry to the project

File: src/components/ItemUnsupported.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  item: DiscriminatedItem;
+}
+
+const ItemUnsupported: FC<ItemunsupportedProps> = ({ item }) => {
+  const { t } = useTranslation();
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {t('The element cannot be opened in the mobile app', {
```
Comment (pyphilia) [inline src/components/ItemUnsupported.tsx:25]:
```suggestion
        {t('The item cannot be opened in the mobile app', {
```

File: src/components/ItemUnsupported.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        raised={true}
+        buttonStyle={{ backgroundColor: PRIMARY_COLOR }}
+        onPress={() =>
+          Sentry.captureException(
+            new Error(`Support for ${item.type} requested`),
+          )
+        }
+        icon={
+          <MaterialIcons
+            name={'feedback'}
```
Comment (pyphilia) [inline src/components/ItemUnsupported.tsx:40]:
```suggestion
            name="feedback"
```

File: src/langs/en.json [hunk 5]
```diff
@@ -87,5 +87,7 @@
   "Add to Bookmarks": "Add to Bookmarks",
   "Remove from Bookmarks": "Remove from Bookmarks",
   "Content": "Content",
-  "Type something....": "Type something...."
+  "Type something....": "Type something....",
+  "The element cannot be opened in the mobile app": "The {{type}} element cannot be opened in the mobile app",
```
Comment (pyphilia) [inline src/langs/en.json:91]:
```suggestion
  "The item cannot be opened in the mobile app": "The {{type}} item cannot be opened in the mobile app",
```

File: src/langs/fr.json [hunk 6]
```diff
@@ -87,5 +87,7 @@
   "Add to Bookmarks": "Ajouter aux Signets",
   "Remove from Bookmarks": "Enlever des Signets",
   "Content": "Contenu",
-  "Type something....": "Ecrire quelque chose...."
+  "Type something....": "Ecrire quelque chose....",
+  "The element cannot be opened in the mobile app": "Cet {{type}} élément ne peut pas être ouvert dans l'application mobile",
```
Comment (pyphilia) [inline src/langs/fr.json:91]:
```suggestion
  "The element cannot be opened in the mobile app": "L'élément {{type}} ne peut pas être ouvert dans l'application mobile",
```

File: src/langs/fr.json [hunk 7]
```diff
@@ -87,5 +87,7 @@
...
+  "I would like to be able to see this item": "J'aimerais pouvoir voir cet article"
```
Comment (pyphilia) [inline src/langs/fr.json:92]:
```suggestion
  "I would like to be able to see this item": "J'aimerais pouvoir voir cet élément"
```
but anyway the message will change (cf comment above)

File: src/components/ItemUnsupported.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {t('The element cannot be opened in the mobile app', {
+          type: item.type,
+        })}
+      </Text>
+      <Button
+        title={t('I would like to be able to see this item')}
```
Comment (pyphilia) [inline src/components/ItemUnsupported.tsx:30]:
```suggestion
        title={t('Send a request to the development team to support itemtype', {type:item.type})}
```

**Review by pyphilia — CHANGES_REQUESTED** [review]

File: src/langs/fr.json [hunk 9]
```diff
@@ -87,5 +87,7 @@
   "Add to Bookmarks": "Ajouter aux Signets",
   "Remove from Bookmarks": "Enlever des Signets",
   "Content": "Contenu",
-  "Type something....": "Ecrire quelque chose...."
+  "Type something....": "Ecrire quelque chose....",
+  "The item cannot be opened in the mobile app": "L'élément {{type}} ne peut pas être ouvert dans l'application mobile",
+  "Send a request to the development team to support": "Envoyer une demande à l'équipe de développement pour soutenir {{type}}"
```
Comment (spaenleh) [inline src/langs/fr.json:92]:
```suggestion
  "Send a request to the development team to support": "Envoyer une demande à l'équipe de développement pour supporter {{type}}"
```

File: src/components/ItemUnsupported.tsx [hunk 10]
```diff
@@ -22,27 +22,31 @@ const ItemUnsupported: FC<ItemunsupportedProps> = ({ item }) => {
...
           Sentry.captureException(
             new Error(`Support for ${item.type} requested`),
-          )
-        }
+          );
+
+          Alert.alert(
+            t('REQUEST_ITEM_SUPPORT_ALERT_TITLE'),
+            t('REQUEST_ITEM_SUPPORT_ALERT_MESSAGE'),
+            [{ text: t('Close'), onPress: () => console.debug('Close alert') }],
```
Comment (spaenleh) [inline src/components/ItemUnsupported.tsx:41]: Is this debugging ?

**Review by spaenleh — APPROVED** [review]

**Pull request merged**